### ISSUE TITLE
On watchOS: drag ended animation starts from full previous screen

### ISSUE THREAD

**Issue by didierburton** [fredyshox/PageView#12]
Not sure if this is due to a new version of watchOS, but `PageScrollState.swift`'s line 70:
`self.pageOffset = 0.0`
seems to reset the position of the "previous" page to its origin before doing the animation to the new page.
As a result, the swipe to an adjacent page has sort of a hiccup.
Moving the statement inside the `DispatchQueue.main.async` block fixes the issue. Lines 69~76 would change as follows:
```
       withAnimation(.easeInOut(duration: 0.2)) {
            self.selectedPage = newPage
        }
        
        DispatchQueue.main.async {
            self.pageOffset = 0.0
            self.isGestureActive = false
        }
```
Even so, swipe "end animations" still do seem a little unnatural, probably because the remaining distance--whatever it might be--is always reached in 0.2 second in an `.easeInOut` animation, which might benefit from being `.easeOut` only.
Thank you.

**Comment by didierburton** [comment]
The fix actually needs to consider whether the new page is different:
The code for a natural gesture (with an `.easeOut`) becomes:
 ```
        withAnimation(.easeOut(duration: 0.2)) {
            if (newPage == selectedPage) {
                self.pageOffset = 0.0
            }
            self.selectedPage = newPage
        }
```
And the `DispatchQueue` block would stay unaffected.

**Comment by fredyshox** [comment]
Sorry for the delay. It seems to work, I will add your fix into the next release :)

**Comment by fredyshox** [comment]
Fixed in 1.5.0
